src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete clone[keys];
+        }
+        break;
+
+      case 'clear':
+        for (const key in clone) {
+          delete clone[key];
+        }
```

Comment:
Instead of additional loop, you can just rewrite you `clone` object
Hint: change clone variable to `let` 
```suggestion
        clone = {}
```